migration/20170614-2-move-third-party-config-to-external-integrations.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+"""Move integration details from the Configuration file into the
+database as ExternalIntegrations
+"""
+import os
+import sys
+import json
+import logging
+from nose.tools import set_trace
+
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, "..")
+sys.path.append(os.path.abspath(package_dir))
+
+from core.model import (
+    ConfigurationSetting,
+    ExternalIntegration as EI,
+    Library,
+    get_one_or_create,
+    production_session,
+)
+
+from api.config import Configuration
+
+log = logging.getLogger(name="Circulation manager configuration import")
+
+def log_import(integration_or_setting, is_new):
+    if is_new:
+        log.info("CREATED: %r" % integration_or_setting)
+    else:
+        log.info("%r already exists." % integration_or_setting)
+
+try:
+    Configuration.load()
+    _db = production_session()
+    LIBRARIES = _db.query(Library).all()
+
+    # Import Circulation Manager base url.
+    circ_manager_conf = Configuration.integration('Circulation Manager')
+    if circ_manager_conf:
+        url = circ_manager_conf.get('url')
+        if url:
+            setting = ConfigurationSetting.sitewide(_db, Configuration.BASE_URL_KEY)
+            is_new = setting.value == None
+            setting.value = unicode(url)
+            log_import(setting, is_new)
+
+    # Import Metadata Wrangler configuration.
+    metadata_wrangler_conf = Configuration.integration('Metadata Wrangler')
+
+    if metadata_wrangler_conf:
+        url = metadata_wrangler_conf.get('url')
+        username = metadata_wrangler_conf.get('client_id')
+        password = metadata_wrangler_conf.get('client_secret')
+
+        integration, is_new = get_one_or_create(
+            _db, EI, protocol=EI.METADATA_WRANGLER, goal=EI.METADATA_GOAL,
+            url=url, username=username, password=password
+        )
+        log_import(integration, is_new)
+
+    # Import Content Server configuration.
+    content_server_conf = Configuration.integration('Content Server')
```

Comment:
I'm not sure where the Content Server falls as an ExternalIntegration, since I think--best-case scenario--it's only used as a license source for certain Collections. I don't see us actually using it for metadata anywhere on circulation, despite the ContentServerCoverageProvider in core.